src/api/calls.rs

```diff
@@ -57,12 +60,16 @@ impl Api {
     where
         Call: subxt::Call + Send + Sync,
     {
+        let mut balance = self
+            .get_balance(&self.signer.account_id().to_ss58check())
```

Comment:
`&self.signer.account_id().to_ss58check()` seems to be called twice in this function. I'd use a variable for it as this call doesn't look as cheap.
```rust 
let signer_address = self.signer.account_id().to_ss58check();
```